lux/action/temporal.py

```diff
@@ -0,0 +1,109 @@
+#  Copyright 2019-2020 The Lux Authors.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+import lux
+from lux.vis.VisList import VisList
+from lux.vis.Vis import Vis
+import pandas as pd
+from lux.core.frame import LuxDataFrame
+
+
+def temporal(ldf):
+    """
+    Generates line charts for temporal fields at different granularities.
+
+    Parameters
+    ----------
+    ldf : lux.core.frame
+            LuxDataFrame with underspecified intent.
+
+    Returns
+    -------
+    recommendations : Dict[str,obj]
+            Object with a collection of visualizations that result from the Temporal action.
+    """
+    visuals = []
+    for c in ldf.columns:
+        if ldf.data_type[c] == "temporal":
+            try:
+                generated_vis = create_vis(ldf, c)
+                visuals.extend(generated_vis)
+            except (ValueError, AttributeError):
+                pass
+    recommendation = {
+        "action": "Temporal",
+        "description": "Show trends over <p class='highlight-descriptor'>time-related</p> attributes.",
+    }
+    # Doesn't make sense to generate a line chart if there is less than 3 datapoints (pre-aggregated)
+    if len(ldf) < 3:
+        recommendation["collection"] = []
+        return recommendation
+    vlist = VisList(visuals)
+    vlist.sort()
+    recommendation["collection"] = vlist
+    return recommendation
+
+
+def create_vis(ldf, col):
+    """
+    Creates and populates Vis objects for the different time-scale visualizations of the provided temporal column.
+
+    Parameters
+    ----------
+    ldf : lux.core.frame
+            LuxDataFrame with underspecified intent.
+
+    col : str
+            Name of temporal column.
+
+    Returns
+    -------
+    visuals : [Vis]
+            Collection of Vis objects.
+    """
+    visuals = []
+    converted_string = ldf[col].astype(str)
+    if converted_string.str.contains("-").any():
```

Comment:
I think this can be done without the regex string by simply [extracting the timescale values from the datetime attribute via Pandas](https://stackoverflow.com/questions/25146121/extracting-just-month-and-year-separately-from-pandas-datetime-column), such as `.dt.year`, `.dt.month`, etc. We should avoid having to do `astype` since this can be an expensive operation.

Reply:
Sounds good, I'll update the action to directly extract the different timescales from the datetime attribute.